src/overlays/actors/ovl_Bg_Spot03_Taki/z_bg_spot03_taki.h

```diff
@@ -6,9 +6,16 @@
 
 struct BgSpot03Taki;
 
+typedef void (*BgSpot03TakiActionFunc)(struct BgSpot03Taki*, struct GlobalContext*);
+
 typedef struct BgSpot03Taki {
-    /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x2C];
+    /* 0x0000 */ DynaPolyActor dyna;
+    /* 0x0164 */ BgSpot03TakiActionFunc actionFunc;
+    /* 0x0168 */ s16 unk_168;
+    /* 0x016A */ u8 unk_16A;
+    /* 0x016C */ u16 switchFlag;
+    /* 0x0170 */ f32 unk_170;
+    /* 0x0174 */ u8 unk_174;
```

Comment:
```suggestion
    /* 0x0168 */ s16 timer;
    /* 0x016A */ u8 state;
    /* 0x016C */ u16 switchFlag;
    /* 0x0170 */ f32 waterfallOpeningAlpha;
    /* 0x0174 */ u8 bufferIndex;
```

Reply:
Additionally, you could create `#define`s for `unk_16A`'s different values, putting them near the top of the .c file. Something like this should do:

    #define STATE_CLOSED 0
    #define STATE_OPENING_IDLE 1
    #define STATE_OPENING_ANIMATED 2
    #define STATE_OPENED 3
    #define STATE_CLOSING 4

Reply:
I'm 99% sure unk_174 is a boolean, especially considering the toggle near the end of Draw:

```c
// ...
    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_bg_spot03_taki.c", 358);

    this->unk_174 = this->unk_174 == 0;
// ...
```

Reply:
this line doesn't represent the whole picture. Take a closer look at where the variable's state is being read:

In `func_808ADEF0` (called by the update function), it's passed into arg1 of `func_808ADAE0`:

    func_808ADAE0(this, this->unk_174);

`func_808ADAE0` uses it's state to select which vtx data to update, choosing either `D_06000800`  (0) or `D_06000990` (1). Later on, the draw routine uses it to select which Vertex list to pass to the RCP for it's next draw task.

    if (this->unk_174 == 0) {
        gSPVertex(oGfxCtx->polyXlu.p++, &D_06000800, 25, 0);
    } else {
        gSPVertex(oGfxCtx->polyXlu.p++, &D_06000990, 25, 0);
    }

Once again, it uses `unk_174` to choose either `D_06000800`  (0) or `D_06000990` (1). It doesn't make sense to assign true/false to these Vtx data.

One last thing is that you might be able to simplify `this->unk_174 = this->unk_174 == 0;` to `this->unk_174 != this->unk_174;`

Reply:
> Once again, it uses unk_174 to choose either D_06000800 (0) or D_06000990 (1). It doesn't make sense to assign true/false to these Vtx data.

It makes perfect sense and you show so in the first sentence. unk_174 is always either 0 or 1 - true or false.

In C, 0 is false and everything that is not 0 is true, which it does consistently throughout the file. If unk_174 is 0, it does stuff with D_06000800, and if unk_174 is anything else, it does stuff with D_06000990. It only ever has two states.

Not only that, but it sets itself to a boolean value in that toggle line I pointed out. `this->unk_174 == 0` evaluates to a boolean value, which it sets unk_174 to be.

I don't get how you can say it doesn't make sense to be a boolean when it only ever has two states and literally sets itself to a boolean value in a function.

Reply:
does it make sense to you to call D_06000800 the false Vtx data, and D_06000990 the true Vtx data?

Reply:
> 
> 
> Additionally, you could create `#define`s for `unk_16A`'s different values, putting them near the top of the .c file. Something like this should do:
> 
> ```
> #define STATE_CLOSED 0
> #define STATE_OPENING_IDLE 1
> #define STATE_OPENING_ANIMATED 2
> #define STATE_OPENED 3
> #define STATE_CLOSING 4
> ```

What about using an enum like @Roman971 [suggested](https://github.com/zeldaret/oot/pull/386#discussion_r495467791) for the Dodongo's Cavern platform?

Reply:
yea, I would do the same.